framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -174,12 +174,93 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
 		}
 	}
 
+	/**
+	 * When there is a failure applying blocks received from the peer,
+	 * it's needede to check whether the tip of the temp block chain has
+	 * preference over the current tip. If so, the temporary chain is restored
+	 * on top of the current chain and the blocks temp table is cleaned up
+	 * @param {ExtendedBlock} lastCommonBlock
+	 * @return {Promise<void>}
+	 * @private
+	 */
+	async _handleBlockProcessingError(lastCommonBlock, peerId) {
+		// If the list of blocks has not been fully applied
+		this.logger.debug('Failed to apply obtained blocks from peer');
+		const [tipBeforeApplying] = await this.storage.entities.TempBlock.get(
+			{},
+			{ sort: 'height:desc', limit: 1, extended: true },
+		);
+
+		const tipBeforeApplyingInstance = await this.processorModule.deserialize(
+			tipBeforeApplying.fullBlock,
+		);
+		// Check if the new tip has priority over the last tip we had before applying
+		const forkStatus = await this.processorModule.forkStatus(
+			this.blocks.lastBlock, // New tip of the chain
+			tipBeforeApplyingInstance, // Previous tip of the chain
+		);
+
+		const newTipHasPreference = forkStatus === FORK_STATUS_DIFFERENT_CHAIN;
```

Comment:
Yes, it is easier to read and to understand what the condition means, especially because the FORK_STATUS_DIFFERENT_CHAIN is not the best name to represent that behavior and I didn't want to change it in this PR